src/DynamoCore/Interfaces/IPathResolver.cs

```diff
@@ -30,6 +30,25 @@ public interface IPathResolver
         /// null.
         /// </summary>
         IEnumerable<string> PreloadedLibraryPaths { get; }
+
+        /// <summary>
+        /// This property represents the root folder where user specific data
+        /// files are stored. If this property returns a null or empty string,
+        /// then PathManager falls back to using %ProgramData%\Dynamo\%version%.
+        /// If this property returns a string that does not represent a valid 
+        /// folder, a DirectoryNotFoundException is thrown.
+        /// </summary>
+        string UserDataRootFolder { get; }
+
+        /// <summary>
+        /// This property represents the root folder where application common 
+        /// data files (i.e. shared among all users on the same machine) are 
+        /// stored. If this property returns a null or empty string, then 
+        /// PathManager falls back to using %AppData%\Dynamo\%version%. If 
+        /// this property returns a string that does not represent a valid 
+        /// folder, a DirectoryNotFoundException is thrown.
+        /// </summary>
+        string CommonDataRootFolder { get; }
```

Comment:
@Benglin it seems wrong to throw exception for these folders because on developer machines where Dynamo isn't installed, these folder don't exist yet and Dynamo would  crash.